apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceState.java

```diff
@@ -230,82 +243,63 @@ public void setSizeLimit(int limit) {
     }
 
     /**
-     * Internal appender uses a per-thread StringBuilder instance to concatenate the tracestate header.
-     * This allows ot limit actual memory usage to be linear to the number of active threads which
-     * is assumed to be far less than the number of active in-flight transactions.
+     * NOT THREAD SAFE! Uses a shared string builder.
+     *
+     * @return the text header generated from {@link #tracestate}.
      */
-    private static class TextTracestateAppender {
-
-        private static final TextTracestateAppender INSTANCE = new TextTracestateAppender();
-        private final ThreadLocal<StringBuilder> tracestateBuffer = new ThreadLocal<StringBuilder>();
-
-        private TextTracestateAppender() {
+    private String generateTracestateHeader() {
```

Comment:
for maintainability, why not make this a synchronized method? It won't impact the sync block it's called in, but will ensure that if someone later adds a call to it elsewhere forgetting to synchronize it, it will be thread-safe